mangum/adapter.py

```diff
@@ -42,24 +52,34 @@ def __init__(
 
         self.app = app
         self.lifespan = lifespan
-        self.api_gateway_base_path = api_gateway_base_path or "/"
-        self.config = LambdaConfig(api_gateway_base_path=self.api_gateway_base_path)
         self.custom_handlers = custom_handlers or []
+        self.config = LambdaConfig(
+            api_gateway_base_path=api_gateway_base_path or "/",
+            text_mime_types=text_mime_types or [*DEFAULT_TEXT_MIME_TYPES],
+        )
+
+    @property
```

Comment:
Can get rid of all of this. 

I think the `api_gateway_base_path` attribute on the adapter instance is something that just got left behind mistakenly in the previous refactor, it should only need to be exposed via the `LambdaConfig` passed to the handlers.

Reply:
What about the `text_mime_types` property? I suggest keeping that one since it's provides a very convenient way to append new text mime types.

Reply:
My main concern is that it introduces additional public API on the adapter class itself which I'd rather just keep in the `config` without committing to an additional property.

I found the original implementation of the `text_mime_types` in [this](https://github.com/jordaneremieff/mangum/blob/245a94bc0400c979f5ff58954673c55baa96481a/mangum/adapter.py#L128) commit, and the way it would work would be that it would combine the user mime types with the defaults.

Could the same behavior solve this here where it isn't `or` but rather `+`?

If that doesn't satisfy the use-case you're thinking, then I would still suggest we still not using the property and instead someone can access `instance.config['text_mime_types']` directly.

Reply:
Regarding appending to user input, I thought about it and I think it would be more annoying than useful. In my opinion, when a user provides a list of text mime types they expect that they are overriding the default. What would happen otherwise is that a user would suddenly discover that some mime types they didn't list are still returned as text, which I would find confusing. Removing default mime types would then become a pain.

I think we should have two behaviors:

- provide a list of mime types in `__init__` - this fully overrides default, i.e. user is fully responsible for declaring all mime types
- modify the list on an instance by appending new values to it

They can get the list of mime types that they can modify (or replace) via `instance.config['text_mime_types']`.

Reply:
Sounds good, let's go with your suggestion, and the `instance.config['text_mime_types']` approach.

Reply:
One more thing: `[*DEFAULT_TEXT_MIME_TYPES]` can just be `[DEFAULT_TEXT_MIME_TYPES]` I think?

Reply:
We should be copying, because if we don't and user modifies the list via config, they would be modifying the global list. This would cause all `Mangum` instances to share the same mime types list. I can hardly imagine a case where you have more than 1 instance of `Mangum`, but nevertheless this is the safest approach.

Reply:
Let's just do `DEFAULT_TEXT_MIME_TYPES` instead of `[*DEFAULT_TEXT_MIME_TYPES]` for now if you've personally not encountered a use-case that would require this.

I agree, I really don't think we'd ever expect there to be more than a single instance given the nature of AWS Lambda, and to anticipate this makes it more confusing from the maintainer perspective if I ever need to revisit this part of the code.

I prefer to assume only a single instance of Mangum is ever being used until some user reports otherwise, then at least we get more details about what's going on and can accommodate them based on more specific information.